### ISSUE TITLE
InvalidOperationException  when calling SendKeys()...

### ISSUE THREAD

**Issue by natiki** [SeleniumHQ/selenium#1860]
## Meta -
OS: Windows 10
Selenium Version: 2.5.3
Browser: PhantonJS 2.1.1
Stack Overflow: http://stackoverflow.com/questions/36129684/selenium-exception-with-sendkeys-to-input-of-type-password

## Expected Behavior -

That SendKeys() places the value into the input control.

## Actual Behavior -

InvalidOperationException  exception when making call to SendKeys() to the password input. Exception is as follows:

> {"errorMessage":"undefined is not an object (evaluating '(C(a)?
> xb(C(a)):window).getComputedStyle(a,null).MozTransform.match')","request":{"headers":
> {"Accept":"application/json, image/png","Connection":"Close","Content-Length":"22","Content-
> Type":"application/json;charset=utf-
> 8","Host":"localhost:60337"},"httpVersion":"1.1","method":"POST","post":"{\"value\":
> [\"something\"]}","url":"/value","urlParsed":
>{"anchor":"","query":"","file":"value","directory":"/","path":"/value","relative":"/value","port":"","host":"","p
> assword":"","user":"","userInfo":"","authority":"","protocol":"","source":"/value","queryKey":
> {},"chunks":["value"]},"urlOriginal":"/session/0c4d6c00-ef54-11e5-9c19-
> 1313e862ecbf/element/:wdc:1458558005770/value"}}

## Steps to reproduce -

- Create a new console application in Visual Studio 2015 using C#
- Install the following NuGet packages using the package manager console
   - Install-Package PhantomJS
   - Install-Package Selenium.WebDriver
- Place the following code into the main application block and exception will be generated on the elPassword.SendKeys() call. The string supplied to SendKeys() is irrelevant.

```c#
PhantomJSOptions options = new PhantomJSOptions();
//IE 11 user agent
options.AddAdditionalCapability("phantomjs.page.settings.userAgent", "Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; rv:11.0) like Gecko");
PhantomJSDriver driver = new PhantomJSDriver(options);
try
{
    //Navigate to the home page
    driver.Navigate().GoToUrl("https://www.advisernet.com.au/avn/logon/logon.jsp");

    //Login
    IWebElement elUserName = driver.FindElement(By.Name("username"));
    elUserName.SendKeys("whatever");
    IWebElement elPassword = driver.FindElement(By.Name("password"));
    elPassword.SendKeys("something");  //**** Exception here *****
}
finally
{
    driver.Quit();
}
```

**Comment by lukeis** [comment]
PhantomJs issues should be logged on the ghostdriver project. Keep in mind that project has become stale and is looking for a new maintainer:
https://github.com/detro/ghostdriver

**Comment by natiki** [comment]
How is an exception inside Selenium WebDriver a PhantomJS issue? And can you recommend an alternative to PhantomJS if you know of any?

**Comment by lukeis** [comment]
the issue you're showing is inside phantomjs / ghostdriver. The selenium library at this point merely sends an HTTP request to the 'ghostdriver' library running inside the phantomjs server which in processing the request responded with the error you saw. Thus, the issue is in the ghostdriver project.

An alternative project? I haven't tried, but this is a new headless driver:
https://github.com/MachinePublishers/jBrowserDriver

**Comment by cgatesman** [comment]
> the issue you're showing is inside phantomjs / ghostdriver. The selenium library at this point merely sends an HTTP request to the 'ghostdriver' library running inside the phantomjs server which in processing the request responded with the error you saw. Thus, the issue is in the ghostdriver project.

The statement above isn't entirely accurate.  The problem is in the javascript webdriver atoms which is part of the Selenium project (https://github.com/SeleniumHQ/selenium/tree/master/javascript/atoms).  These atoms are borrowed from a specific Selenium version/build and imported into GhostDriver.  In turn, they are imported to phantomjs through a specific version/build of GhostDriver.  Ultimately, these are maintained by the Selenium project, so, technically, this is a Selenium issue.  However, it is entirely possible this may not be a problem with the latest Selenium.  

Here is the current offering from PhantomJS:

PhantomJS 2.1.1 -> GhostDriver 1.2.0 which uses Selenium 2.39.0 WebDriver Atoms

The real problem may be 2 fold:

1. GhostDriver needs to update the Selenium WebDriver Atoms it is shipped with.
2. PhantomJS needs to update to that newer version of GhostDriver.

IMO, this 3 layered separate projects design has been a detriment to the whole Selenium/PhantomJS eco system and one of the main reasons it hasn't thrived.  It's too bad.

**Comment by cgatesman** [comment]
FWIW, I found that this issue is resolved if you use the latest 2.0.0 GhostDriver with PhantomJS (see comments in https://github.com/detro/ghostdriver/issues/534).
